### ISSUE TITLE
[Mac] "Do you want to accept incoming connections?"

### ISSUE THREAD

**Issue by vitobotta** [zadam/trilium#509]
HI! I just downloaded the app for Mac, and was presented with the dialog in the subject. Why would the app need incoming connections? Thanks

**Comment by zadam** [comment]
Hello, I haven't seen this in mac yet. But it might be something to do with the fact that this app is essentially http server + web app frontend, so mac warns about the http server part. But it's exposed locally only, not to any network ...

**Comment by meichthys** [comment]
I also get this every time I launch the app:
<img width="532" alt="Screen Shot 2020-05-04 at 6 52 12 PM" src="https://user-images.githubusercontent.com/10717998/81021099-5f83b000-8e38-11ea-92b8-e9e254a74ae6.png">

**Comment by zadam** [comment]
It looks like this is caused by the fact Trilium is unsigned - https://superuser.com/questions/912656/how-do-i-stop-my-mac-from-asking-to-accept-incoming-network-connections

**Comment by meichthys** [comment]
TLDR: Just run `sudo codesign --force --deep --sign - /path/to/Trilium\ Notes.app` and relaunch Trilium, and `Allow` Trilium through the firewall. After that you won't receive the firewall prompt after each launch.

It seems reasonable that OSX would prompt about an unsigned application, but Apple's documentation makes it sound like unsigned apps that are allowed through the firewall will be signed by the local machine and then allowed:

> If you run an unsigned app that is not listed in the firewall list, a dialog appears with options to Allow or Deny connections for the app. If you choose Allow, OS X signs the application and automatically adds it to the firewall list. If you choose Deny, OS X adds it to the list but denies incoming connections intended for this app.

I can verify that OSX initially sees Trilium as unsigned:

<img width="618" alt="Screen Shot 2020-05-05 at 7 38 06 AM" src="https://user-images.githubusercontent.com/10717998/81062145-78757b00-8ea3-11ea-8f90-d296ef340b1e.png">

But after I allow it through the firewall, it should be 'signed' by OSX according to the above documentation - and this isn't happening. However, after self-signing the application myself and clicking `Allow` once more on the firewall prompt, I no longer receive the popup each time I launch Trilium:

<img width="1004" alt="Screen Shot 2020-05-05 at 7 47 43 AM" src="https://user-images.githubusercontent.com/10717998/81063156-50871700-8ea5-11ea-9e4a-7eb83fde661a.png">

**Comment by meichthys** [comment]
A quick update on this...
It looks like after each reboot, it is still necessary to "Allow" Trilium once. Subsequent app launches will not prompt again until a reboot.
